modules/data.land/R/match_pft.R

```diff
@@ -0,0 +1,53 @@
+##' Match model PFTs
+##' 
+##' Matches BETYdb species IDs to model-specific PFTs
+##' 
+##' @param bety_species_id  vector of BETYdb species IDs
+##' @param pfts             settings$pfts.  List of pfts with database matching based on name
+##' @param con              database connection
+##' 
+##' @return table of BETYdb PFT IDs matched to species IDs
+##' 
+match_pft <- function(bety_species_id, pfts, con){
+  
+  ### get species to PFT mappting
+  query <- NULL
+  for (pft in pfts) {
+    if (is.null(query)) {
+      query <- paste0("SELECT bp.id as bety_pft_id, bp.name as pft, bs.id as bety_species_id, bs.scientificname as latin FROM pfts as bp INNER JOIN ", 
+                      "pfts_species AS bps ON bps.pft_id = bp.id INNER JOIN species AS bs ON bs.id = bps.specie_id WHERE ", 
+                      "bp.name = '", pft$name, "'")
+    } else {
+      query <- paste0(query, " OR bp.name = '", pft$name, "'")
+    }
+  }
+  translation <- db.query(query, con = con)
+  
+  ## Check for duplicate bety_species_ids in PFTs
+  bad <- translation[duplicated(pfts$bety_species_id),]
```

Comment:
I think this was supposed to be `translation[duplicated(translation$bety_species_id),]`